controllers/pkg/reconcilers/repository/reconciler.go

```diff
@@ -119,7 +119,7 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		// Delete the repo from the git server
 		// when successful remove the finalizer
 		if cr.Spec.Lifecycle.DeletionPolicy == commonv1alpha1.DeletionDelete {
-			if err := r.deleteRepo(ctx, giteaClient, cr); err != nil {
+			if err := r.deleteRepo(ctx, r.giteaClient, cr); err != nil {
```

Comment:
Not part of this PR (no need to address), but line 108 is weird, why doesn't it just check r.giteaClient directly? I don't think the Get() method should be needed. But maybe a follow up PR (with the change in this line that will become a short PR).

Reply:
Yes, I agree. I'll do this in the next PR.